format_all.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MPL-2.0
+
+WORKSPACE_ROOT=$(pwd)
+
+EXCLUDED_CRATES=$(sed -n '/^\[workspace\]/,/^\[.*\]/{/exclude = \[/,/\]/p}' Cargo.toml | grep -v "exclude = \[" | tr -d '", \]')
+
+echo "Formatting workspace"
+cargo fmt
+
+for CRATE in $EXCLUDED_CRATES; do
+    CRATE_DIR="$WORKSPACE_ROOT/$CRATE"
+
+    if [ -d "$CRATE_DIR" ]; then
+        echo "Formatting excluded crate: $CRATE"
+        (cd "$CRATE_DIR" && cargo fmt)
+    else
+        echo "Directory for crate $CRATE does not exist"
+    fi
+done
+
+cd -
```

Comment:
So what's the point of entering the previous directory at the end of the script?